### ISSUE TITLE
Building z3 for python 3 fails

### ISSUE THREAD

**Issue by swflint** [NixOS/nixpkgs#55591]
## Issue description

Running `nix-env -iA nixpkgs.python36Packages.z3` on unstable gives the following:

```
replacing old 'z3-4.8.4'
installing 'z3-4.8.4'
these derivations will be built:
  /nix/store/82r6d0kfs4ddad4nd9dxakkbadyr1a1r-z3-4.8.4.drv
building '/nix/store/82r6d0kfs4ddad4nd9dxakkbadyr1a1r-z3-4.8.4.drv'...
unpacking sources
unpacking source archive /nix/store/0xv697d7qqmnw3vczflbhag1dlkfx32d-source
source root is source
patching sources
configuring
Failed to read file src/ast/recfun_decl_plugin.h

opt = --prefix, arg = /nix/store/hl9f9vli9vwizn071h3xj8wqqdg557p6-z3-4.8.4
opt = --python, arg =
opt = --pypkgdir, arg = /nix/store/hl9f9vli9vwizn071h3xj8wqqdg557p6-z3-4.8.4/lib/python3.6/site-packages
New component: 'util'
New component: 'polynomial'
New component: 'sat'
New component: 'nlsat'
New component: 'lp'
New component: 'hilbert'
New component: 'simplex'
New component: 'automata'
New component: 'interval'
New component: 'realclosure'
New component: 'subpaving'
New component: 'ast'
New component: 'rewriter'
New component: 'macros'
New component: 'normal_forms'
New component: 'model'
New component: 'tactic'
New component: 'substitution'
New component: 'parser_util'
New component: 'grobner'
New component: 'euclid'
New component: 'core_tactics'
New component: 'proofs'
New component: 'solver'
New component: 'sat_tactic'
New component: 'arith_tactics'
New component: 'nlsat_tactic'
New component: 'subpaving_tactic'
New component: 'aig_tactic'
New component: 'ackermannization'
New component: 'cmd_context'
New component: 'smt2parser'
New component: 'fpa'
New component: 'pattern'
New component: 'bit_blaster'
New component: 'smt_params'
New component: 'proto_model'
New component: 'smt'
New component: 'bv_tactics'
New component: 'fuzzing'
New component: 'smt_tactic'
New component: 'sls_tactic'
New component: 'qe'
New component: 'sat_solver'
New component: 'fd_solver'
New component: 'muz'
New component: 'dataflow'
New component: 'transforms'
New component: 'rel'
New component: 'spacer'
New component: 'clp'
New component: 'tab'
New component: 'ddnf'
New component: 'bmc'
New component: 'fp'
New component: 'ufbv_tactic'
New component: 'smtlogic_tactics'
New component: 'fpa_tactics'
New component: 'portfolio'
New component: 'opt'
New component: 'api'
New component: 'extra_cmds'
New component: 'shell'
New component: 'test'
New component: 'api_dll'
New component: 'dotnet'
New component: 'dotnetcore'
New component: 'java'
New component: 'ml'
New component: 'cpp'
Python bindings directory was detected.
New component: 'python'
New component: 'python_install'
New component: 'js'
New component: 'cpp_example'
New component: 'z3_tptp'
New component: 'c_example'
New component: 'maxsat'
New component: 'dotnet_example'
New component: 'java_example'
New component: 'ml_example'
New component: 'py_example'
Generating src/util/z3_version.h from src/util/z3_version.h.in
Generated 'src/util/z3_version.h'
Generating src/api/dotnet/Properties/AssemblyInfo.cs from src/api/dotnet/Properties/AssemblyInfo.cs.in
Generating src/api/dotnet/Properties/AssemblyInfo.cs from src/api/dotnet/Properties/AssemblyInfo.cs.in
Generated 'src/tactic/smtlogics/qfufbv_tactic_params.hpp'
Generated 'src/tactic/sls/sls_params.hpp'
Generated 'src/muz/base/fp_params.hpp'
Generated 'src/sat/sat_params.hpp'
Generated 'src/sat/sat_asymm_branch_params.hpp'
Generated 'src/sat/sat_params.hpp'
Generated 'src/sat/sat_asymm_branch_params.hpp'
Generated 'src/sat/sat_simplifier_params.hpp'
Generated 'src/sat/sat_scc_params.hpp'
Generated 'src/util/lp/lp_params.hpp'
Generated 'src/opt/opt_params.hpp'
Generated 'src/math/polynomial/algebraic_params.hpp'
Generated 'src/math/realclosure/rcf_params.hpp'
Generated 'src/solver/combined_solver_params.hpp'
Generated 'src/solver/parallel_params.hpp'
Generated 'src/ackermannization/ackermannize_bv_tactic_params.hpp'
Generated 'src/ackermannization/ackermannization_params.hpp'
Generated 'src/smt/params/smt_params_helper.hpp'
Generated 'src/parsers/util/parser_params.hpp'
Generated 'src/model/model_params.hpp'
Generated 'src/model/model_evaluator_params.hpp'
Generated 'src/nlsat/nlsat_params.hpp'
Generated 'src/ast/pp_params.hpp'
Generated 'src/ast/fpa/fpa2bv_rewriter_params.hpp'
Generated 'src/ast/normal_forms/nnf_params.hpp'
Generated 'src/ast/rewriter/rewriter_params.hpp'
Generated 'src/ast/rewriter/bool_rewriter_params.hpp'
Generated 'src/ast/rewriter/arith_rewriter_params.hpp'
Generated 'src/ast/rewriter/array_rewriter_params.hpp'
Generated 'src/ast/rewriter/bv_rewriter_params.hpp'
Generated 'src/ast/rewriter/fpa_rewriter_params.hpp'
Generated 'src/ast/rewriter/poly_rewriter_params.hpp'
Generated 'src/ast/pattern/pattern_inference_params_helper.hpp'
Generated 'src/ast/pattern/database.h'
Component api
Component portfolio
Component smtlogic_tactics
Component ackermannization
Component model
Component rewriter
Component ast
Component util
Component polynomial
Component automata
Component solver
Component tactic
Component proofs
Component sat_solver
Component core_tactics
Component macros
Component normal_forms
Component aig_tactic
Component bv_tactics
Component bit_blaster
Component arith_tactics
Component sat
Component sat_tactic
Component nlsat_tactic
Component nlsat
Component smt_tactic
Component smt
Component cmd_context
Component proto_model
Component smt_params
Component pattern
Component smt2parser
Component parser_util
Component substitution
Component grobner
Component euclid
Component simplex
Component fpa
Component lp
Component fp
Component muz
Component qe
Component clp
Component transforms
Component hilbert
Component dataflow
Component tab
Component rel
Component bmc
Component fd_solver
Component ddnf
Component spacer
Component ufbv_tactic
Component fpa_tactics
Component sls_tactic
Component subpaving_tactic
Component subpaving
Component interval
Component realclosure
Component opt
Component extra_cmds
Component shell
Traceback (most recent call last):
  File "scripts/mk_make.py", line 17, in <module>
    mk_auto_src()
  File "/build/source/scripts/mk_util.py", line 3017, in mk_auto_src
    mk_all_install_tactic_cpps()
  File "/build/source/scripts/mk_util.py", line 3136, in mk_all_install_tactic_cpps
    mk_install_tactic_cpp(cnames, c.src_dir)
  File "/build/source/scripts/mk_util.py", line 3125, in mk_install_tactic_cpp
    generated_file = mk_genfile_common.mk_install_tactic_cpp_internal(h_files_full_path, path)
  File "/build/source/scripts/mk_genfile_common.py", line 723, in mk_install_tactic_cpp_internal
    raise e
  File "/build/source/scripts/mk_genfile_common.py", line 700, in mk_install_tactic_cpp_internal
    for line in fin:
  File "/nix/store/whd8n5gwyn8c7i9nlp1w01dxkq79kwbp-python3-3.6.8/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 1809: ordinal not in range(128)
builder for '/nix/store/82r6d0kfs4ddad4nd9dxakkbadyr1a1r-z3-4.8.4.drv' failed with exit code 1
error: build of '/nix/store/82r6d0kfs4ddad4nd9dxakkbadyr1a1r-z3-4.8.4.drv' failed
```

### Steps to reproduce

`nix-env -iA nixpkgs.python36Packages.z3`

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.15, NixOS, 19.03pre166449.be445a9074f (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(swflint): `"nixpkgs-19.03pre169106.1a88aa9e0cd, nixpkgs-unstable-19.03pre169106.1a88aa9e0cd, nixpkgs1809-18.09.2132.109a28ab954"`
 - channels(root): `"nixos-19.03pre166449.be445a9074f, nix1709-17.09.3269.14f9ee66e63, nix1803-18.03.133389.b551f89e256, nix1809-18.09.1829.0396345b794, unstable-19.03pre164715.201d739b0ff"`
 - nixpkgs: `/home/swflint/.nix-defexpr/channels/nixpkgs`

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

**Comment by andersk** [comment]
Fixed upstream: Z3Prover/z3#2131.

**Comment by swflint** [comment]
I believe #55769 closes this.
